aws-sam-cli.hcl

```diff
@@ -0,0 +1,24 @@
+description = "The AWS Serverless Application Model (SAM) CLI is an open-source CLI tool that helps you develop serverless applications containing Lambda functions, Step Functions, API Gateway, EventBridge, SQS, SNS and more."
+test = "sam --version"
+repository = "https://github.com/aws/aws-sam-cli"
+
+darwin {
+  source = "https://github.com/aws/aws-sam-cli/releases/download/v${version}/aws-sam-cli-macos-arm64.pkg"
```

Comment:
This won't work on amd64 macs?

Reply:
results of test:
`hermit-packages🐚 gaumann@gaumann-macbookpro hermit-packages % hermit test aws-sam-cli --trace
info:validate: Validating aws-sam-cli-1.115.0
trace: HEAD https://github.com/aws/aws-sam-cli/releases/download/v1.115.0/aws-sam-cli-linux-x86_64.zip
trace: HEAD https://objects.githubusercontent.com/github-production-release-asset-2e65be/92205085/e1ebfe8a-407d-4c3f-bc3d-2a2758c48d7a?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=AKIAVCODYLSA53PQK4ZA%2F20240417%2Fus-east-1%2Fs3%2Faws4_request&X-Amz-Date=20240417T040611Z&X-Amz-Expires=300&X-Amz-Signature=4783cfc44121639880a38fdd957ae1f7cc170a505341f406ae3eac33e8bc53f8&X-Amz-SignedHeaders=host&actor_id=0&key_id=0&repo_id=92205085&response-content-disposition=attachment%3B%20filename%3Daws-sam-cli-linux-x86_64.zip&response-content-type=application%2Foctet-stream
trace: HEAD https://github.com/aws/aws-sam-cli/releases/download/v1.115.0/aws-sam-cli-macos-arm64.pkg
trace: HEAD https://objects.githubusercontent.com/github-production-release-asset-2e65be/92205085/d9479f2d-a75e-423a-b0f8-acd734bac5dd?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=AKIAVCODYLSA53PQK4ZA%2F20240417%2Fus-east-1%2Fs3%2Faws4_request&X-Amz-Date=20240417T040612Z&X-Amz-Expires=300&X-Amz-Signature=a3d7e18832ebc9057ac3ed5480e2bb23141f13b8e8a1ce7afbc57a86bc2bfee1&X-Amz-SignedHeaders=host&actor_id=0&key_id=0&repo_id=92205085&response-content-disposition=attachment%3B%20filename%3Daws-sam-cli-macos-arm64.pkg&response-content-type=application%2Foctet-stream
trace: HEAD https://github.com/aws/aws-sam-cli/releases/download/v1.115.0/aws-sam-cli-macos-arm64.pkg
trace: HEAD https://objects.githubusercontent.com/github-production-release-asset-2e65be/92205085/d9479f2d-a75e-423a-b0f8-acd734bac5dd?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=AKIAVCODYLSA53PQK4ZA%2F20240417%2Fus-east-1%2Fs3%2Faws4_request&X-Amz-Date=20240417T040612Z&X-Amz-Expires=300&X-Amz-Signature=a3d7e18832ebc9057ac3ed5480e2bb23141f13b8e8a1ce7afbc57a86bc2bfee1&X-Amz-SignedHeaders=host&actor_id=0&key_id=0&repo_id=92205085&response-content-disposition=attachment%3B%20filename%3Daws-sam-cli-macos-arm64.pkg&response-content-type=application%2Foctet-stream
info:validate: Validating aws-sam-cli@latest
trace: HEAD https://github.com/aws/aws-sam-cli/releases/download/v1.115.0/aws-sam-cli-linux-x86_64.zip
trace: HEAD https://objects.githubusercontent.com/github-production-release-asset-2e65be/92205085/e1ebfe8a-407d-4c3f-bc3d-2a2758c48d7a?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=AKIAVCODYLSA53PQK4ZA%2F20240417%2Fus-east-1%2Fs3%2Faws4_request&X-Amz-Date=20240417T040611Z&X-Amz-Expires=300&X-Amz-Signature=4783cfc44121639880a38fdd957ae1f7cc170a505341f406ae3eac33e8bc53f8&X-Amz-SignedHeaders=host&actor_id=0&key_id=0&repo_id=92205085&response-content-disposition=attachment%3B%20filename%3Daws-sam-cli-linux-x86_64.zip&response-content-type=application%2Foctet-stream
trace: HEAD https://github.com/aws/aws-sam-cli/releases/download/v1.115.0/aws-sam-cli-macos-arm64.pkg
trace: HEAD https://objects.githubusercontent.com/github-production-release-asset-2e65be/92205085/d9479f2d-a75e-423a-b0f8-acd734bac5dd?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=AKIAVCODYLSA53PQK4ZA%2F20240417%2Fus-east-1%2Fs3%2Faws4_request&X-Amz-Date=20240417T040612Z&X-Amz-Expires=300&X-Amz-Signature=a3d7e18832ebc9057ac3ed5480e2bb23141f13b8e8a1ce7afbc57a86bc2bfee1&X-Amz-SignedHeaders=host&actor_id=0&key_id=0&repo_id=92205085&response-content-disposition=attachment%3B%20filename%3Daws-sam-cli-macos-arm64.pkg&response-content-type=application%2Foctet-stream
trace: HEAD https://github.com/aws/aws-sam-cli/releases/download/v1.115.0/aws-sam-cli-macos-arm64.pkg
trace: HEAD https://objects.githubusercontent.com/github-production-release-asset-2e65be/92205085/d9479f2d-a75e-423a-b0f8-acd734bac5dd?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=AKIAVCODYLSA53PQK4ZA%2F20240417%2Fus-east-1%2Fs3%2Faws4_request&X-Amz-Date=20240417T040612Z&X-Amz-Expires=300&X-Amz-Signature=a3d7e18832ebc9057ac3ed5480e2bb23141f13b8e8a1ce7afbc57a86bc2bfee1&X-Amz-SignedHeaders=host&actor_id=0&key_id=0&repo_id=92205085&response-content-disposition=attachment%3B%20filename%3Daws-sam-cli-macos-arm64.pkg&response-content-type=application%2Foctet-stream
info:validate: Validating aws-sam-cli@1
trace: HEAD https://github.com/aws/aws-sam-cli/releases/download/v1.115.0/aws-sam-cli-linux-x86_64.zip
trace: HEAD https://objects.githubusercontent.com/github-production-release-asset-2e65be/92205085/e1ebfe8a-407d-4c3f-bc3d-2a2758c48d7a?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=AKIAVCODYLSA53PQK4ZA%2F20240417%2Fus-east-1%2Fs3%2Faws4_request&X-Amz-Date=20240417T040611Z&X-Amz-Expires=300&X-Amz-Signature=4783cfc44121639880a38fdd957ae1f7cc170a505341f406ae3eac33e8bc53f8&X-Amz-SignedHeaders=host&actor_id=0&key_id=0&repo_id=92205085&response-content-disposition=attachment%3B%20filename%3Daws-sam-cli-linux-x86_64.zip&response-content-type=application%2Foctet-stream
trace: HEAD https://github.com/aws/aws-sam-cli/releases/download/v1.115.0/aws-sam-cli-macos-arm64.pkg
trace: HEAD https://objects.githubusercontent.com/github-production-release-asset-2e65be/92205085/d9479f2d-a75e-423a-b0f8-acd734bac5dd?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=AKIAVCODYLSA53PQK4ZA%2F20240417%2Fus-east-1%2Fs3%2Faws4_request&X-Amz-Date=20240417T040612Z&X-Amz-Expires=300&X-Amz-Signature=a3d7e18832ebc9057ac3ed5480e2bb23141f13b8e8a1ce7afbc57a86bc2bfee1&X-Amz-SignedHeaders=host&actor_id=0&key_id=0&repo_id=92205085&response-content-disposition=attachment%3B%20filename%3Daws-sam-cli-macos-arm64.pkg&response-content-type=application%2Foctet-stream
trace: HEAD https://github.com/aws/aws-sam-cli/releases/download/v1.115.0/aws-sam-cli-macos-arm64.pkg
trace: HEAD https://objects.githubusercontent.com/github-production-release-asset-2e65be/92205085/d9479f2d-a75e-423a-b0f8-acd734bac5dd?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=AKIAVCODYLSA53PQK4ZA%2F20240417%2Fus-east-1%2Fs3%2Faws4_request&X-Amz-Date=20240417T040612Z&X-Amz-Expires=300&X-Amz-Signature=a3d7e18832ebc9057ac3ed5480e2bb23141f13b8e8a1ce7afbc57a86bc2bfee1&X-Amz-SignedHeaders=host&actor_id=0&key_id=0&repo_id=92205085&response-content-disposition=attachment%3B%20filename%3Daws-sam-cli-macos-arm64.pkg&response-content-type=application%2Foctet-stream
info:validate: Validating aws-sam-cli@1.115
trace: HEAD https://github.com/aws/aws-sam-cli/releases/download/v1.115.0/aws-sam-cli-linux-x86_64.zip
trace: HEAD https://objects.githubusercontent.com/github-production-release-asset-2e65be/92205085/e1ebfe8a-407d-4c3f-bc3d-2a2758c48d7a?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=AKIAVCODYLSA53PQK4ZA%2F20240417%2Fus-east-1%2Fs3%2Faws4_request&X-Amz-Date=20240417T040611Z&X-Amz-Expires=300&X-Amz-Signature=4783cfc44121639880a38fdd957ae1f7cc170a505341f406ae3eac33e8bc53f8&X-Amz-SignedHeaders=host&actor_id=0&key_id=0&repo_id=92205085&response-content-disposition=attachment%3B%20filename%3Daws-sam-cli-linux-x86_64.zip&response-content-type=application%2Foctet-stream
trace: HEAD https://github.com/aws/aws-sam-cli/releases/download/v1.115.0/aws-sam-cli-macos-arm64.pkg
trace: HEAD https://objects.githubusercontent.com/github-production-release-asset-2e65be/92205085/d9479f2d-a75e-423a-b0f8-acd734bac5dd?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=AKIAVCODYLSA53PQK4ZA%2F20240417%2Fus-east-1%2Fs3%2Faws4_request&X-Amz-Date=20240417T040612Z&X-Amz-Expires=300&X-Amz-Signature=a3d7e18832ebc9057ac3ed5480e2bb23141f13b8e8a1ce7afbc57a86bc2bfee1&X-Amz-SignedHeaders=host&actor_id=0&key_id=0&repo_id=92205085&response-content-disposition=attachment%3B%20filename%3Daws-sam-cli-macos-arm64.pkg&response-content-type=application%2Foctet-stream
trace: HEAD https://github.com/aws/aws-sam-cli/releases/download/v1.115.0/aws-sam-cli-macos-arm64.pkg
trace: HEAD https://objects.githubusercontent.com/github-production-release-asset-2e65be/92205085/d9479f2d-a75e-423a-b0f8-acd734bac5dd?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=AKIAVCODYLSA53PQK4ZA%2F20240417%2Fus-east-1%2Fs3%2Faws4_request&X-Amz-Date=20240417T040612Z&X-Amz-Expires=300&X-Amz-Signature=a3d7e18832ebc9057ac3ed5480e2bb23141f13b8e8a1ce7afbc57a86bc2bfee1&X-Amz-SignedHeaders=host&actor_id=0&key_id=0&repo_id=92205085&response-content-disposition=attachment%3B%20filename%3Daws-sam-cli-macos-arm64.pkg&response-content-type=application%2Foctet-stream
debug:aws-sam-cli-1.115.0:exec: /Users/gaumann/Library/Caches/hermit/pkg/aws-sam-cli-1.115.0/aws-sam-cli/sam --version
debug: SAM CLI, version 1.115.0
hermit-packages🐚 (py3) gaumann@gaumann-macbookpro hermit-packages % sam --version
SAM CLI, version 1.98.0`

Reply:
Yes, but it has to work on amd64 macs as well. Hermit's three required platforms are darwin/arm64, darwin/amd64 and linux/amd64.

Reply:
I have added a package for amd64 macs. Also corrected hash of linux package.